sjsonnet/src/sjsonnet/Parser.scala

```diff
@@ -291,6 +291,20 @@ object Parser{
         exprs(preLocals.length)
       val postLocals = exprs.drop(preLocals.length+1).takeWhile(_.isInstanceOf[Expr.Member.BindStmt])
         .map(_.asInstanceOf[Expr.Member.BindStmt])
+      
+      /* 
+       * Prevent duplicate fields in list comprehension. See: https://github.com/databricks/sjsonnet/issues/99
+       * 
+       * If comps._1 is a forspec with value greater than one lhs cannot be a Expr.Str
+       * Otherwise the field value will be overriden by the multiple iterations of forspec
+       */
+      if (lhs.isInstanceOf[Expr.Str] &&
+          comps._1.isInstanceOf[Expr.ForSpec] && 
+          comps._1.asInstanceOf[Expr.ForSpec].cond.isInstanceOf[Expr.Arr] &&
+          comps._1.asInstanceOf[Expr.ForSpec].cond.asInstanceOf[Expr.Arr].value.length > 1
+      ) {
+        Fail.opaque(s"""no duplicate field: "${lhs.asInstanceOf[Expr.Str].value}" """)
```

Comment:
This probably can be more nicely written as:

```scala
(lhs, comps) match {
  case (Expr.Str(_, _), (_, Expr.ForSpec(_, _, Expr.Arr(_, values)) ) if values.length > 1 =>  Fail.opaque(...)
  case _ => // do nothing
}
```

Reply:
Good call, updated